### PR TITLE
Add support for JDK 10 in both Maven and Gradle

### PR DESCRIPTION
**Pull request by cheenamalhotra**
_This PR:_
- Builds jre10 jars for the driver, replacing jre9 jars.
- Updates Appveyor to use SQL Server 2017 for testing.
- Updates Appveyor to run with JDK 10
- Updates Travis CI to run with JDK 10

_Also Updates Maven Plugin:_ 
- jacoco-maven-plugin : **0.8.1**

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 1]
```diff
@@ -3227,17 +3227,8 @@ public PreparedStatement prepareStatement(String sql,
...
-        
-        // Make sure SQLServerPreparedStatement42 is used for 4.2 and above. 
-        if (Util.use42Wrapper() || Util.use43Wrapper()) {
-            st = new SQLServerPreparedStatement42(this, sql, resultSetType, resultSetConcurrency,
-                    SQLServerStatementColumnEncryptionSetting.UseConnectionSetting);
-        }
-        else {
-            st = new SQLServerPreparedStatement(this, sql, resultSetType, resultSetConcurrency,
+        PreparedStatement st = new SQLServerPreparedStatement(this, sql, resultSetType, resultSetConcurrency,
                     SQLServerStatementColumnEncryptionSetting.UseConnectionSetting);
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java]: un-indent this

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 2]
```diff
@@ -3274,17 +3257,8 @@ public CallableStatement prepareCall(String sql,
...
-
-        // Make sure SQLServerCallableStatement42 is used for 4.2 and above. 
-        if (Util.use42Wrapper() || Util.use43Wrapper()) {
-            st = new SQLServerCallableStatement42(this, sql, resultSetType, resultSetConcurrency,
-                    SQLServerStatementColumnEncryptionSetting.UseConnectionSetting);
-        }
-        else {
-            st = new SQLServerCallableStatement(this, sql, resultSetType, resultSetConcurrency,
+        CallableStatement st = new SQLServerCallableStatement(this, sql, resultSetType, resultSetConcurrency,
                     SQLServerStatementColumnEncryptionSetting.UseConnectionSetting);
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java]: same here

File: build.gradle [hunk 3]
```diff
@@ -27,11 +27,11 @@ allprojects {
 
 if (!hasProperty('buildProfile') || (hasProperty('buildProfile') && buildProfile == "build43")){
 	
-	jreVersion = "jre9"
```
Comment (peterbae) [inline build.gradle]: Change commenting on line 5 to java 10

File: pom.xml [hunk 4]
```diff
@@ -256,46 +258,46 @@
...
- 						<id>attach-javadocs</id>
- 						<goals>
- 							<goal>jar</goal>
- 						</goals>
- 					</execution>
- 				</executions>
- 			</plugin>
+				<artifactId>maven-javadoc-plugin</artifactId>
+				<version>3.0.0</version>
+
```
Comment (rene-ye) [inline pom.xml]: Weird place to have a return

File: pom.xml [hunk 5]
```diff
@@ -256,46 +258,46 @@
...
+				<configuration>
+					<failOnError>true</failOnError>
+				</configuration>
+
```
Comment (rene-ye) [inline pom.xml]: Same as above, why is this config sandwiched by newline.

**Comment by softboy99** [comment]
waiting for good news

**Pull request merged**